power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -250,6 +414,21 @@ template <dataset_type_tag dataset_type_> class Dataset {
         buffers_.push_back(Buffer{});
     }
 
+    template <class RangeType>
+    RangeType get_span_impl(Buffer const& buffer, ComponentInfo const& info, Idx scenario,
+                            RangeType const& total_range) const {
```

Comment:
it's common to start with the most important argument, then other relevant arguments, and then metadata. Compare e.g. in Python: positional arguments followed by arguments (`*args`) and then keyword arguments (`**kwargs`)

```suggestion
    RangeType get_span_impl(RangeType const& total_range, Idx scenario, IdxVector const& indptr, ComponentInfo const& info) const {
```

Note: not relevant for this PR but if you have one or more output targets (captured by non-const `&`), then it's common to put them last in the list